torchtune/models/clip/_position_embeddings.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+
+import torch
+from torch import nn
+
+# TODO (Felipe): add load hooks + interpolation on positional encodings,
+# so max_num_tiles can be variable and a trained model can be adapted to a
+# new value.
+
+
+class TokenPositionalEmbedding(nn.Module):
+    """
+    Token positional embedding for images. Different for every token (patch) in an image.
+
+    Args:
+        embed_dim (int): The dimensionality of each token embedding.
+        patch_grid_size (int): The side of a squared grid that represents how many
+             patches are in one tile, i.e. tile_size // patch_size.
+    """
+
+    def __init__(self, embed_dim: int, patch_grid_size: int) -> None:
+        super().__init__()
+        scale = embed_dim**-0.5
+        self.positional_embedding = nn.Parameter(
+            scale
+            * torch.randn((patch_grid_size**2 + 1, embed_dim))  # +1 for CLS token
+        )
+
+    def forward(self, x: torch.Tensor, *args) -> torch.Tensor:
+        """
+        args:
+            x (torch.Tensor): Tensor with shape (*, n_tokens, embed_dim)
+        returns:
+            torch.Tensor: The input tensor with added positional embeddings.
+        """
+        return x + self.positional_embedding
+
+
+class TiledTokenPositionalEmbedding(nn.Module):
+    """
+    Token positional embedding for tiled images. There are two positional embeddings in this module:
+        - local_token_positional_embedding: same for every tile, different for every token. Equivalent to
+            ``torchtune.models.clip._position_embeddings.TokenPositionalEmbedding``, but gated.
+        - global_token_positional_embedding: different for every tile, different for every token.
+
+    Notice that tile is different from patch (token). An image is divided into tiles during pre-processing,
+    and patches are the outcome of the convolution in the ViT applied to each tile.
```

Comment:
I feel like this can also be replicated at the top of the file and expanded a bit to explain this relationship better